Sources/Sentry/SentryDependencyContainer.m

```diff
@@ -347,4 +348,42 @@ - (SentryMXManager *)metricKitManager SENTRY_THREAD_SANITIZER_DOUBLE_CHECKED_LOC
 
 #endif // SENTRY_HAS_METRIC_KIT
 
+#if SENTRY_HAS_REACHABILITY
+- (SentryReachability *)reachability SENTRY_DISABLE_THREAD_SANITIZER(
+    "double-checked lock produce false alarms")
+{
+    SENTRY_LAZY_INIT(_reachability, [[SentryReachability alloc] init]);
+}
+#endif // !TARGET_OS_WATCH
+
+- (SentryScopeContextPersistentStore *)scopeContextStore SENTRY_THREAD_SANITIZER_DOUBLE_CHECKED_LOCK
+{
+    SENTRY_LAZY_INIT(_scopeContextStore,
+        [[SentryScopeContextPersistentStore alloc] initWithFileManager:self.fileManager]);
+}
+
+#if SENTRY_HAS_UIKIT
+- (SentryWatchdogTerminationBreadcrumbProcessor *)
+    getWatchdogTerminationBreadcrumbProcessorWithMaxBreadcrumbs:(NSInteger)maxBreadcrumbs
+{
+    // This method is only a factory, therefore do not keep a reference.
+    // The processor will be created each time it is needed.
+    @synchronized(sentryDependencyContainerInstanceLock) {
```

Comment:
`m`: I think you could ditch the `@synchronized` then or consider not adding the method at all.

Reply:
Why remove it? You (the previous maintainers) introduced the `SentryDispatchFactory` also as a factory to create instances instead of `SentryDispatchQueueWrapper`

Reply:
Because we don't need the `@synchronized`; it's obsolete. You can safely access this method from multiple threads. Each thread gets a different instance of `SentryWatchdogTerminationBreadcrumbProcessor` and the `[self fileManager]` is thread safe. Unless I'm missing something.

Reply:
> introduced the SentryDispatchFactory also as a factory to create instances instead of SentryDispatchQueueWrapper

Mainly for testing purposes. I don't mind keeping it, no strong opinion.